core/os/android/adb/commands.go

```diff
@@ -372,6 +372,35 @@ func (b *binding) QueryPerfettoServiceState(ctx context.Context) (*device.Perfet
 	return result, nil
 }
 
+// TODO: Currently using Android Q/10, API 29 as ANGLE support cut-off
+// TODO: If I knew squat about go then I would refactor this to be shared
+//  function w/ "SupportsPerfetto" and have const values defined somewhere
+//  such as "PERFETTO_SUPPORT_OS_VERSION = 28" and "ANGLE_SUPPORT_OS_VERSION = 29"
+//  then call that shared version with the version as input
+func (b *binding) SupportsAngle(ctx context.Context) bool {
+	os := b.Instance().GetConfiguration().GetOS()
+	return os.GetAPIVersion() >= 29
```

Comment:
Sorry for jumping into a WIP patch. Just in case I forgot. Should we just use 30(R)? Supporting one version back while we are not really actively testing it may end up biting us. Besides all quarterly releases of Q are now closed :)

Reply:
I'm still doing all of my testing on Q so I'll keep it Q for now. Can potentially bump it up if/when all ANGLE engineers have moved to R.